src/App.jsx

```diff
@@ -14,33 +16,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_BY_NAME = 'name';
+const SORT_BY_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function preparedGoodsData(goods, sortValue, reversed) {
+  const preparedData = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortValue) {
+    preparedData.sort((good1, good2) => {
+      switch (sortValue) {
+        case SORT_BY_NAME:
+          return good1.localeCompare(good2);
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        case SORT_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    preparedData.reverse();
+  }
+
+  return preparedData;
+}
+
+export const App = () => {
+  const [sortValue, setSortValue] = useState('');
+  const [reversed, setReversed] = useState(false);
+
+  const visibleGoods = preparedGoodsData(goodsFromServer, sortValue, reversed);
+
+  const sortState = sortValue || reversed;
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  function isReversed() {
+    setReversed(yes => !yes);
+  }
+
+  function isReset() {
+    setSortValue('');
+    setReversed(false);
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortValue(SORT_BY_NAME)}
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortValue !== SORT_BY_NAME,
+          })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortValue(SORT_BY_LENGTH)}
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortValue !== SORT_BY_LENGTH,
+          })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => isReversed()}
+          type="button"
+          className={cn('button is-success', {
```

Comment:
Looks like u copied this class from another element. Two buttons have the same color 
![image](https://github.com/user-attachments/assets/26dc52b3-ae2d-4e00-9794-6da15aa5a4b9)

Reply:
> Looks like u copied this class from another element. Two buttons have the same color ![image](https://private-user-images.githubusercontent.com/120860193/369387098-26dc52b3-ae2d-4e00-9794-6da15aa5a4b9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcwMzA3MDEsIm5iZiI6MTcyNzAzMDQwMSwicGF0aCI6Ii8xMjA4NjAxOTMvMzY5Mzg3MDk4LTI2ZGM1MmIzLWFlMmQtNGUwMC05Nzk0LTZkYTE1YWE1YTRiOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyMlQxODQwMDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mODA4ZmY3YmI0MGU3NjZiODA2ZTk2ODY0NmI2ZTc1YjE0ZGI1YzdlNTJkMWM0YmRhZTQ4MTgxY2E0OWJlZTQ4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.WANotKhWBgIp2T6DmQoHa2-gHnWkmWYiD4nHXG0aLMk)

Good point